### DISCUSSION TITLE
Open/close paradigm in Stitches (React)

### DISCUSSION THREAD

**Discussion by erasebegin** [stitchesjs/stitches#1128] [Help] [answered]
I have a modal that opens/closes according to a boolean value. In styled components I would just pass in a prop called something like `isOpen` which would contain the boolean piece of state and then use that `isOpen` value throughout the styled component to toggle various css properties.
 
There's also the sucky CSS way to do it which is to dynamically add and remove a class, which is what currently exists and which I am trying to convert to Stitches. Stitches is dynamic enough that I could still use this method to toggle the modal, but it's not very clean. I was wondering what the Stitches way to handle this situation would be?

 🙏🙏🙏

**Comment by wolffbruno** [comment, marked as answer]
Boolean variant can't help you?

`const Button = styled('button', {
  // base styles

  variants: {
    outlined: {
      true: {
        borderColor: 'lightgray',
      },
    },
  },
});

() => <Button outlined>Button</Button>;`

**Reply by erasebegin** [reply]
Ahh, I forgot about this possibility for accepting booleans. That will work, thank you!
